.buildkite/pipeline.elastic-agent-package.yml

```diff
@@ -105,6 +105,58 @@ steps:
         artifact_paths:
           - "build/distributions/**/*"
 
+      - label: ":package: Cross Building and package FIPS linux/amd64 elastic-agent"
+        key: package-elastic-agent_fips
+        agents:
+          provider: "gcp"
+          machineType: "c2-standard-16"
+          diskSizeGb: 400
+        env:
+          PLATFORMS: "linux/amd64"
+          FIPS: true
+        command: |
+          if [[ -z "$${MANIFEST_URL}" ]]; then
+            export MANIFEST_URL=$(buildkite-agent meta-data get MANIFEST_URL --default "")
+            if [[ -z "$${MANIFEST_URL}" ]]; then
+              echo ":broken_heart: Missing MANIFEST_URL variable or empty string provided"
+              exit 1
+            fi
+          fi
+          if [[ -z "$${MAGEFILE_VERBOSE}" ]]; then
+            export MAGEFILE_VERBOSE=$(buildkite-agent meta-data get MAGEFILE_VERBOSE --default "0")
+          fi
+          .buildkite/scripts/steps/package.sh
```

Comment:
Can we use a script for this? Maybe I'm bias, but I prefer pipelines to call scripts instead, what do you think?

Maybe ` .buildkite/scripts/steps/package.sh` could contain those things?

Reply:
I am sure it could but that is a decision more on the eng-prod side of things.
For this PR I only duplicated the FIPS package steps exactly as the non-FIPS ones.
I personally don't use `.buildkite/scripts/steps/package.sh` for packaging locally, so I guess it impacts only CI pipelines.

Reply:
> I am sure it could but that is a decision more on the eng-prod side of things.

We facilitate, so in the end, it's a system that evolves over time and is maintained by many people. 

That's the reason I suggested removing the duplicated code, Can we do it within this PR?

Reply:
@pchila can't you use matrix to the existing packaging step and just expose the appropriate FIPS env var there, one with false and another with true? 🙂

Reply:
@pkoutsovasilis that is possible if we accept extra PLATFORM in the FIPS step (it won't break anything as packaging code already manage the correct platform list for FIPS packages, but it may be confusing to people)

Reply:
@pchila for now I deem that this approach seems acceptable to me for now. @v1v wdyt?

Reply:
pushed a new commit with matrix implementation 3ecfcae5ceeccde2b658921d322de585b0624b81

Reply:
Thanks @pchila and @pkoutsovasilis for the suggestion and changes